packages/headless/src/api/knowledge/knowledge-api-answer.ts

```diff
@@ -0,0 +1,177 @@
+import {SearchRequest} from '../search/search/search-request';
+import {knowledgeApi} from './knowledge-api';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+interface GeneratedAnswerStream {
+  answerStyle: string | undefined;
+  contentFormat: GeneratedContentFormat | undefined;
+  answer: string | undefined;
+  citations: GeneratedAnswerCitation[] | undefined;
+  generated: boolean;
+  isStreaming: boolean;
+  isLoading: boolean;
+}
+
+interface Data {
+  answerStyle: string | undefined;
+  contentFormat: GeneratedContentFormat | undefined;
+  answer: string | undefined;
+  citations: GeneratedAnswerCitation[] | undefined;
+  generated: boolean;
+  isStreaming: boolean;
+  isLoading: boolean;
+}
+
+const updateCacheWithEvent = (event: EventSourceMessage, draft: Data) => {
+  const message = JSON.parse(event.data);
+  const payload = JSON.parse(message.payload);
+  if (message.payloadType === 'genqa.headerMessageType') {
+    draft.answerStyle = payload.answerStyle;
+    draft.contentFormat = payload.contentFormat;
+    draft.isStreaming = true;
+    draft.isLoading = false;
+  }
+
+  if (message.payloadType === 'genqa.messageType') {
+    if (draft.answer === undefined) {
+      draft.answer = payload.textDelta;
+    } else {
+      draft.answer = draft.answer?.concat(payload.textDelta);
+    }
+  }
+
+  if (message.payloadType === 'genqa.citationsType') {
+    draft.citations = payload.citations;
+  }
+
+  if (message.payloadType === 'genqa.endOfStreamType') {
+    draft.generated = Boolean(payload.answerGenerated);
+    draft.isStreaming = false;
+  }
+};
+
+const onOpenStream = async (response: Response) => {
+  if (
+    response.ok &&
+    response.headers.get('content-type')?.includes('text/event-stream')
+  ) {
+    return;
+  }
+
+  const isClientSideError =
+    response.status >= 400 && response.status < 500 && response.status !== 429;
+
+  if (isClientSideError) {
+    throw new FatalError({
+      message: 'Error opening stream',
+      code: response.status,
+    });
+  } else {
+    throw new Error();
+  }
+};
+
+const onError = (err: Error) => {
+  let retryCount = 0;
+
+  if (err instanceof FatalError) {
+    throw err;
+  }
+
+  if (++retryCount > 3) {
+    const error = {
+      message: 'Failed to complete stream.',
+      code: 1,
+    };
+    throw new FatalError(error);
+  }
+};
+
+export const answerAPI = knowledgeApi.injectEndpoints({
+  overrideExisting: true,
+  endpoints: (builder) => ({
+    getAnswer: builder.query<GeneratedAnswerStream, Partial<SearchRequest>>({
+      queryFn: () => ({
+        data: {
+          answerStyle: undefined,
+          contentFormat: undefined,
+          answer: undefined,
+          citations: undefined,
```

Comment:
why define the `undefined` ?

Reply:
This is the initial object the query needs. And at first those values are undefined